### PR TITLE
Package update + more docs on dev setup

### PR DESCRIPTION
**Pull request by dmchoiboi**
# Pull Request Summary

_What is this PR changing? Why is this change being made? Any caveats you'd like to highlight? Link any relevant documents, links, or screenshots here if applicable._

* Upgrade poetry to 1.8.2 to fix package version conflicts with pip-tools in requirements-dev.txt
* Add docs on setting to `WORKSPACE` env var to be able to run tests locally
* Fix faulty `type: ignore` and handle type narrowing explicitly
* Add test case for high priority + zipartifact codepath for test coverage

## Test Plan and Usage Guide

_How did you validate that your PR works correctly? How do you run or demo the code? Provide enough detail so a reviewer can reasonably reproduce the testing procedure. Paste example command line invocations if applicable._
Tested pip install + mypy + pytest on a fresh venv using python 3.8.10

```
# new venv
python -m venv venvs/llm-engine
source ./venvs/llm-engine/bin/activate

# install packages
pip install -r ../requirements-dev.txt
pip install -r requirements.txt && \
    pip install -r requirements-test.txt && \
    pip install -r requirements_override.txt && \
    pip install -e .

# typecheck
mypy . --install-types

# unit tests
pytest
```

### CODE REVIEW

**Review by seanshi-scale — APPROVED** [review]
lgtm if integration tests pass, looks like some code coverage tests failed? guess we'll need to fix that too

File: model-engine/README.md [hunk 1]
```diff
@@ -41,3 +41,9 @@ Run `mypy . --install-types` to set up mypy.
 
 Most of the business logic in Model Engine should contain unit tests, located in
 [`tests/unit`](./tests/unit). To run the tests, run `pytest`.
+
+Make sure to set a WORKSPACE env var that points to the root directory of this repo
```
Comment (yunfeng-scale) [inline model-engine/README.md:45]: do we actually need this?

Comment (dmchoiboi) [reply model-engine/README.md:45]: the code requires a WORKSPACE env to be set for things to work.

File: model-engine/model_engine_server/infra/services/live_endpoint_builder_service.py [hunk 2]
```diff
@@ -494,7 +494,9 @@ def get_base_image_params(
 
         # The context should be whatever WORKDIR is in the container running the build app itself.
         inference_folder = "model-engine/model_engine_server/inference"
-        base_path: str = os.getenv("WORKSPACE")  # type: ignore
+        base_path = os.getenv("WORKSPACE")
```
Comment (dmchoiboi) [inline model-engine/model_engine_server/infra/services/live_endpoint_builder_service.py:497]: going to extract this outside of the method invocation

File: model-engine/README.md (hunk 1 above)
Comment (dmchoiboi) [reply model-engine/README.md:45]: nevermind, i decided to just add a mock for the unit test case

Comment (dmchoiboi) [reply model-engine/README.md]: Nevermind, decided to add mocks to handle it in the unittest directly

File: model-engine/tests/unit/conftest.py [hunk 3]
```diff
@@ -3383,6 +3383,48 @@ def build_endpoint_request_async_custom(
...
+            last_updated_at=datetime(2022, 1, 4),
+            metadata={},
+            creation_task_id="test_creation_task_id",
+            endpoint_type=ModelEndpointType.ASYNC,
+            destination="test_destination",
+            status=ModelEndpointStatus.READY,
+            current_model_bundle=model_bundle_3,
+            owner=test_api_key,
+        ),
+        high_priority=True,
```
Comment (dmchoiboi) [inline model-engine/tests/unit/conftest.py:3405]: Add this fixture to test the high_priority code path

**Pull request merged**